manifests/agent.pp

```diff
@@ -291,23 +291,40 @@
 
   # Ensure that the correct config file is used.
   if $manage_startup_script {
-    zabbix::startup { $servicename:
-      pidfile                   => $pidfile,
-      agent_configfile_path     => $agent_configfile_path,
-      zabbix_user               => $zabbix_user,
-      additional_service_params => $additional_service_params,
-      service_type              => $service_type,
-      service_name              => 'zabbix-agent',
-      require                   => Package[$zabbix_package_agent],
+    if $agent_configfile_path =~ /agent2/ {
```

Comment:
please set a variable within this if block for the service_name and don't duplicate the zabbix::startup block.

Reply:
I initially used $servicename for the service_name parameter, but $servicename varies for different systems (which is set in params.pp), and the service_name variable is used for getting the template file name. I'll probably have to modify startup.pp to fix it.

Reply:
Nvm, I get it now. I'll fix it shortly